src/forms/inputs/textarea.js

```diff
@@ -8,19 +8,19 @@ import { compose } from '../../utils'
 /**
  *
  * A textarea input that can be used in a `redux-forms`-controlled form. Optionally displays a character count.
- * 
+ *
  * @name Textarea
  * @type Function
  * @param {Object} input - A `redux-forms` [input](http://redux-form.com/6.5.0/docs/api/Field.md/#input-props) object
  * @param {Object} meta - A `redux-forms` [meta](http://redux-form.com/6.5.0/docs/api/Field.md/#meta-props) object
- * @param {Number} [maxLength=500] - The maximum allowed length of the input
+ * @param {Number} [maxLength=500] - The maximum allowed length of the input. Accepts a boolean, e.g., `false` for the option to not set a max length.
```

Comment:
You can put `{Number|Boolean}` as the type. Also, you should just mention `false` rather than saying "a boolean" because the latter suggests that `true` is an acceptable value.